src/Model/JoinLinkTrait.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+trait JoinLinkTrait
+{
+    /**
+     * The short name of the join link.
+     *
+     * @var string
+     */
+    protected $join;
+
+    public function getJoin(): Join
+    {
+        return $this->getOwner()->getElement($this->join);
+    }
+
+    public function hasJoin(): bool
+    {
+        return $this->join && $this->getOwner()->hasElement($this->join);
```

Comment:
```suggestion
        return $this->join !== null;
```

getter should fail if name is set but Join does not exist!